### PR TITLE
Small container improvement

### PR DESCRIPTION
**Pull request by FabianSchuetze**
@kunaltyagi commented in the (unrelated) PR #4719 that a `std::array` could replace a c-style array. I am just filing this short PR because I think the changed indeed improves the code.

### CODE REVIEW

File: gpu/containers/src/initialization.cpp [hunk 1]
```diff
@@ -115,16 +116,13 @@ namespace
...
             {0x10,   8}, {0x11,   8}, {0x12,   8}, {0x13,  8}, {0x20,  32}, {0x21, 48}, {0x30, 192},
-            {0x35, 192}, {0x50, 128}, {0x52, 128}, {0x53, 128}, {0x60, 64}, {0x61, 128}, {-1, -1}
-        };
-        int index = 0;
-        while (gpuArchCoresPerSM[index].SM != -1) 
-        {
-            if (gpuArchCoresPerSM[index].SM == ((major << 4) + minor) ) 
-                return gpuArchCoresPerSM[index].Cores;
-            index++;
+            {0x35, 192}, {0x50, 128}, {0x52, 128}, {0x53, 128}, {0x60, 64}, {0x61, 128}
```
Comment (kunaltyagi) [inline gpu/containers/src/initialization.cpp:121]: We should add more here. 62, 70, 72, 75, 80, 86 are the new architectures (Pascal, Volta, Turing, Ampere)

Comment (larshg) [reply gpu/containers/src/initialization.cpp:121]: Seems a lot like this: https://github.com/NVIDIA/cuda-samples/blob/master/Common/helper_cuda.h#L652-L668.

Comment (FabianSchuetze) [reply gpu/containers/src/initialization.cpp]: Thank you very much, @kunaltyagi  and @larshg, for your comments! Your comments were of great help to me! I added the new architectures. I am still wondering about two aspects:
- Don't we support two Kepler architectures? We are missing values of 32 and 37 in comparison to the link Lars provided. Plus, we mention older architectures (until 30) in the list. Do we know if our code compiles with these architectures? Should we maybe remove these values? 
- Why do our original code and the NVIDIA code check for {-1,-1} at the end. Isn't that redundant?

Comment (kunaltyagi) [reply gpu/containers/src/initialization.cpp]: * Based on this (https://en.wikipedia.org/wiki/CUDA#GPUs_supported), we could add the older versions, but we'd have to remove them again when CUDA 11 becomes the oldest supported version. We can add 37 (but not before 35)
* That's not needed anymore thanks to your refactoring

Comment (kunaltyagi) [reply gpu/containers/src/initialization.cpp]: Our oldest CUDA support is for CUDA 9 I believe. @larshg can we remove pre 3.0?

Comment (larshg) [reply gpu/containers/src/initialization.cpp]: We don't compile for less than 3.0 as of our CMake https://github.com/PointCloudLibrary/pcl/blob/master/cmake/pcl_find_cuda.cmake#L33-L39.

So I don't see any reason not to remove pre 3.0.

Comment (FabianSchuetze) [reply gpu/containers/src/initialization.cpp]: Thank you again, @kunaltyagi  and @larshg, for your feedback! I harmonized the information in `pcl_find_cuda` and `initialization.cpp`. All compute capability versions starting from 3.0 are now included. Do you agree with the changes?

**Review by kunaltyagi — APPROVED** [review]

**Comment by kunaltyagi** [comment]
I approved the PR, but the CI is not happy about the decomposition in the for loop (C++17 feature). Could you please make one final change? :)

**Comment by FabianSchuetze** [comment]
Of course, @kunaltyagi! I was already wondering if I am permitted to use c++17. The standard in the Cmakelists is set to 14. Just out of curiosity, could this maybe be updated? The oldest compiler used in the build pipeline seems to be gcc7 which supports all but one PR of c++17.  Anyway, thank you again very much for your support!

File: gpu/containers/src/initialization.cpp [hunk 2]
```diff
@@ -39,6 +39,8 @@
 
 #include "cuda.h"
 #include <cstdio>
+#include <array>
+#include <algorithm>
```
Comment (kunaltyagi) [inline gpu/containers/src/initialization.cpp:43]: Why is this needed? Please add `// for ...` in the added headers

Comment (FabianSchuetze) [reply gpu/containers/src/initialization.cpp]: Oh, bother! Sorry - leaving the algo header was a mistake. I replaced the raw loop with a `std::find_if` upon the advice of my linter. I tried it but didn't want to push the result and left the include by mistake. If you agree with the linter's sentiment, I can do that again. Does PCL have any code policy about that?

**Review by kunaltyagi — APPROVED** [review]

**Review by larshg — APPROVED** [review]

File: gpu/containers/src/initialization.cpp (hunk 2 above)
Comment (kunaltyagi) [reply gpu/containers/src/initialization.cpp]: Not really. In that situation, a return is easier to read.
